pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1082,6 +1083,50 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "setSyncSetFailedCondition should be present",
+			existing: []runtime.Object{
+				testInstalledClusterDeployment(time.Now()),
+				createSyncSetInstanceObj(hivev1.ApplyFailureSyncCondition),
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
+				cond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.SyncSetFailedCondition)
+				if cond == nil {
+					assert.NotNil(t, cond, "missing SyncSetFailedCondition status condition")
+				} else {
+					assert.Equal(t, corev1.ConditionTrue, cond.Status, "did not get expected state for SyncSetFailedCondition condition")
+				}
+			},
+		},
+		{
+			name: "setSyncSetFailedCondition value should be corev1.ConditionFalse",
+			existing: []runtime.Object{
+
+				func() runtime.Object {
+					cd := testInstalledClusterDeployment(time.Now())
+					var conditions []hivev1.ClusterDeploymentCondition
+					cd.Status.Conditions = append(
+						conditions,
```

Comment:
Append to `cd.Status.Conditions` instead of creating an empty slice to which to append. If you really want to ensure that `cd.Status.Conditions` contains *only* the one condition, then create a slice with that one condition and don't use append at all.
```
cd.Status.Conditions = append(cd.Status.Conditions, hivev1.ClusterDeploymentCondition{...})
```
or
```
cd.Stauts.Conditions = []hivev1.ClusterDeploymentCondition{{...}}
```